src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs

```diff
@@ -79,18 +64,36 @@ public async Task<LeaseResource> ReadOrCreateLeaseResource(string name)
             }
             
         }
+        
+        private async Task<BlobContainerClient> ContainerClient()
+        {
+            var serviceClient = _settings.AzureCredential != null && _settings.ServiceEndpoint != null
+                ? new BlobServiceClient(_settings.ServiceEndpoint, _settings.AzureCredential, _settings.BlobClientOptions)
+                : new BlobServiceClient(_settings.ConnectionString);
+            
+            var client = serviceClient.GetBlobContainerClient(_settings.ContainerName);
+            
+            // Make sure that `CreateIfNotExistsAsync()` only get called once for every AzureApi instance
+            if (!_initialized)
+            {
+                await client.CreateIfNotExistsAsync();
+                _initialized = true;
+            }
+            
+            return client;
+        }
```

Comment:
Actual fix, make sure that a new container client is instantiated every time a container client is requested, but only try to initialize the azure container once for every instance of `AzureApiImpl`